backend-implicaction/src/test/java/com/dynonuggets/refonteimplicaction/controller/JobsPostingControllerTest.java

```diff
@@ -126,4 +126,61 @@ void getJobByIdWithoutJwtShouldBeForbidden() throws Exception {
 
         verify(jobPostingService, times(0)).getJobById(anyLong());
     }
+
+    @Test
+    @WithMockUser
+    void archiveJobShouldChangeStatus() throws Exception {
+        //given
+        JobPostingDto givenDto = JobPostingDto.builder()
+                .id(1L)
+                .isArchive(false)
+                .build();
+
+        JobPostingDto expectedDto = JobPostingDto.builder()
+                .id(1L)
+                .isArchive(true)
+                .build();
+
+        given(jobPostingService.toggleArchiveJobPosting(anyLong())).willReturn(expectedDto);
+
+        //when
+        final ResultActions resultActions = mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, givenDto.getId()).contentType(APPLICATION_JSON));
+
+        //then
+        resultActions.andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is((givenDto.getId().intValue()))))
+                .andExpect(jsonPath("$.archive", is(expectedDto.isArchive())));
+        verify(jobPostingService, times(1)).toggleArchiveJobPosting(anyLong());
+    }
+
+    @Test
+    @WithMockUser
+    void archiveJobPostingWhithUnexistingIdShouldThrowException() throws Exception {
+        //given
+        final long jobId = 123L;
+        final NotFoundException exception = new NotFoundException(String.format(JOB_NOT_FOUND_MESSAGE, jobId));
+        given(jobPostingService.toggleArchiveJobPosting(anyLong())).willThrow(exception);
+
+        //when
+        ResultActions resultActions = mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobId));
+
+        //then
+        resultActions.andExpect(status().isNotFound());
+        verify(jobPostingService, times(1)).toggleArchiveJobPosting(anyLong());
+    }
+
+    @Test
+    void archiveJobByIdWithoutJwtShouldBeForbidden() throws Exception {
+        //given
+        final long jobId = 123L;
+
+        //when
+        ResultActions resultActions = mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, jobId).contentType(APPLICATION_JSON));
+
+        //then
+        resultActions.andDo(print());
+        resultActions.andExpect(status().isForbidden());
```

Comment:
tu peux chaîner les méthodes : 
```suggestion
        resultActions.andDo(print())
            .andExpect(status().isForbidden());
```